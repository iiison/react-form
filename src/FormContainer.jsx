import React, { Component } from 'react'
import PropTypes            from 'prop-types' // eslint-disable-line import/no-extraneous-dependencies

import validations          from './utils/validationRules'

/**
 * Select rule from 3 diffrent places(predefined rules, form level rules or input level rules)
 */
export function selectRule({ ruleValue, customFormRules, customRules }) {
  return validations[ruleValue] || customRules[ruleValue] || customFormRules[ruleValue]
}

/**
 * Validates if the the selected rule(returned from `selectRule`) is valid or not.
 * if rule is valid then tests the input field value agains the selected rule.
 * returns empty string if rule is valid and saticefies the condition.
 */
export function validateRule({ id, value, displayName, ruleValue, ruleArgs, validation }) {
  if (validation) {
    const stringifiedValue = value.toString()
    const isRuleSatisfied = ( ruleValue !== 'required' && !stringifiedValue )
      ? true
      : validation.rule.apply(null, ruleArgs).test(value.toString())
    let error = ''

    if (!isRuleSatisfied) {
      error = validation.formatter.apply(null, [displayName || id, ...ruleArgs])

      return error
    }
  } else {
    throw `invalid validation rule: ${ruleValue}, please use an existing validation rule name or pass a custom function with same name through 'customRules' prop in Input: ${id}. Rule value should be an object with keys: 'rule' as an Regex and 'formatter' as a function, that formats the value.` // eslint-disable-line
  }

  return ''
}


/**
 * Main Component, a HOC, will store all fields' state.
 */
export default class FormContainer extends Component {
  constructor(props) {
    super(props)

    const {
      isDisabled = false,
      shouldValidateForm,
      defaultClasses
    } = this.props

    this.state = {
      shouldValidateForm,
      defaultClasses,
      isDisabled,
      errors     : {},
      isFetching : false
    }
  }

  getChildContext() {
    return {
      setFieldValue : ({ event, field, value, isMultipleValues, id = 'value' }, callBack) => {
        const fieldName = field.id
        const setState = (name, fieldValue) => {
          this.setState((prevState) => ({
            ...prevState,
            fields : {
              ...prevState.fields,
              [fieldName] : {
                ...prevState.fields[name],
                shouldValidateField : true,
                [id]                : fieldValue
              }
            }
          }), callBack)
        }

        if (!value && value !== '') {
          setState(fieldName, event.currentTarget.value)
        } else if (isMultipleValues) {
          const stateCopy = { ...this.state }

          for (const item in value) {
            stateCopy.fields[item].shouldValidateField = true
            stateCopy.fields[item][id] = value[item]
          }

          this.setState((prevState) => ({
            ...prevState,
            ...stateCopy
          }))
        } else {
          setState(fieldName, value)
        }

        return this.state
      },

      addField : (data) => {
        this.setState((prevState) => ({
          ...prevState,
          errors : {
            ...prevState.errors,
            [data.id] : ''
          },
          fields : {
            ...prevState.fields,
            [data.id] : {
              ...data,
              // shouldValidateField : false // Not sure why this was added
            }
          }
        }))
      },

      setFormData : (data) => {
        this.setState((prevState) => ({
          ...prevState,
          ...data
        }))
      },

      validateForm : (fieldName) => {
        const { shouldValidateForm, fields } = this.state
        const field = fields[fieldName]

        if (shouldValidateForm) {
          if (field) {
            if ( field.shouldValidateField ) {
              this.validateField(field)
            }

            return
          }

          for (const unit in fields) {
            const fieldData = fields[unit]

            if (fieldData) {
              this.validateField(fieldData)
            }
          }
        }

        return
      },
      formData : this.state
    }
  }

  /**
   * React Lifecycle Method: Renders the data
   *
   * @return {DOM} Main container DOM.
   */
  render() {
    const { children } = this.props

    return (
      <form action=''>
        {children}
      </form>
    )
  }

  validateField = ({
    id,
    value,
    validate,
    displayName,
    customRules = {}
  }) => {
    let error = ''
    const rules = validate ? validate.split('|') : ''
    const { customValidators : customFormRules } = this.props

    if (rules.length) {
      for (const rule in rules) {
        const ruleName = rules[rule]
        const ruleDetails = ruleName.split('-')
        const [ruleValue, ...ruleArgs] = ruleDetails
        const validation = selectRule({ ruleValue, customFormRules, customRules })

        try {
          error = validateRule({ id, value, displayName, ruleValue, ruleArgs, validation })

          if (error !== '') {
            break
          }
        } catch (invalidRuleError) {
          throw invalidRuleError
        } 
      }

      this.setState((prevState) => ({
        ...prevState,
        errors : {
          ...prevState.errors,
          [id] : error
        }
      }))
    }
  }

  static defaultProps = {
    errors             : {}, // eslint-disable-line
    children           : <div />,
    isDisabled         : false,
    shouldValidateForm : true,
    customValidators   : {},
    defaultClasses     : {
      contClass  : '',
      fieldClass : '',
      errorClass : '',
      labelClass : ''
    }
  }

  static propTypes = {
    children           : PropTypes.node,
    isDisabled         : PropTypes.bool,
    shouldValidateForm : PropTypes.bool,
    customValidators   : PropTypes.object, // eslint-disable-line
    defaultClasses     : PropTypes.shape({
      labelClass : PropTypes.string,
      contClass  : PropTypes.string,
      errorClass : PropTypes.string,
      fieldClass : PropTypes.string,
    })
  }

  static childContextTypes = {
    addField      : PropTypes.func,
    setFieldValue : PropTypes.func,
    validateForm  : PropTypes.func,
    formData      : PropTypes.object, // eslint-disable-line
    setFormData   : PropTypes.func
  }
}

