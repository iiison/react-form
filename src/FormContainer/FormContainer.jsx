import React, { Component } from 'react'

import { Form } from '$COMPONENTS'
import validations from '$UTILS/validationRules'

/**
 * Test all validation rules ofa field
 *
 * @param {Object} fieldData all field realated data
 *
 * @return {String}         validation error ina field
 */
export function testFieldRules(fieldData) {
  const { validate, value, displayName } = fieldData

  const validationRules = validate.split('|')

  for (const validation in validationRules) {
    const { rule, formatter } = validations[validationRules[validation]]()

    const result = rule.test(value)

    if (result === false) {
      return formatter(displayName)
    }
  }
}

/**
 * Generic Form Handler, Handles form state,
 * Pass Form elemnt's JSON throught the FormContainer and
 * it will redner the form.
 */
export default class FormContainer extends Component {
  /**
   * Constructor for the container
   *
   * @param {Object} props form data
   */
  constructor(props) {
    super(props)
    const { formData : { fields } } = props

    this.state = {
      isTested : false,
    }

    this.state.fields = fields.reduce((prev, curr) => {
      const temp = {
        value          : '',
        error          : '',
        shouldBeTested : false,
        ...curr
      }

      return {
        ...prev,
        [temp.id] : temp
      }
    }, {})
  }

  /**
   * DOM Event : Chaanges `shouldBeTested`, will be executed on
   * keyUp in an input field
   *
   * @param {String} field field name
   */
  makeFieldTestable = (field) => {
    const fieldRef = this.state.fields[field]

    fieldRef.shouldBeTested = true

    this.setState((previousState) => {
      return {
        fields : {
          ...previousState.fields,
          [field] : fieldRef
        }
      }
    })
  }

  /**
   * Handle Key press event on the input,
   * tests field value if field is already has error.
   * @param  {String} field Field name in which event was started
   */
  handleKeyPress = (field) => {
    const fieldData = this.state.fields[field]
    const { error } = fieldData

    this.makeFieldTestable(field)

    if (error) {
      this.validateInput(field)
    }
  }

  /**
   * DOM Event: Handles input value change
   *
   * @param {Object} event DOM event object
   */
  handleChange = (event) => {
    event.preventDefault()

    const elemRef = event.target
    const idName = elemRef.id

    this.setState((previousState) => {
      const { fields } = previousState

      return {
        fields : {
          ...fields,
          [idName] : {
            ...fields[idName],
            value : elemRef.value
          }
        }
      }
    })
  }

  /**
   * Validate single input
   *
   * @param  {String}  field       field name to be tested
   * @param  {Boolean} isFinalTest will be true if whole form is being tested
   * @return {String}              validation error of the field
   */
  validateInput = (field, isFinalTest) => {
    const fieldData = this.state.fields[field]
    const { validate } = fieldData

    if ((fieldData.shouldBeTested === true || isFinalTest) && validate) {
      const error = testFieldRules(fieldData) || ''

      this.setState((previousState) => {
        const { fields } = previousState
        const updatedField = {
          ...fields[field],
          error
        }

        return {
          fields : {
            ...fields,
            [field] : updatedField
          }
        }
      })

      return error
    }
  }

  /**
   * Handles Form Submission, callback actual form submit fxn
   * after the form validation. validation is the part of plugin.
   *
   * @param {Object} event DOM Event object
   */
  handleFormSubmit = (event) => {
    event.preventDefault()

    const { handleSubmit } = this.props
    const { fields } = this.state
    const errors = []

    for ( const field in fields ) {
      const error = this.validateInput(field, true)

      error && errors.push(error)
    }

    if (!errors.length) {
      handleSubmit(fields)
    }
  }

  /**
   * React Lifecycle Method: Renders the data
   *
   * @return {DOM} Main container DOM.
   */
  render() {
    const { fields } = this.state
    const {
      error,
      messages,
      isFetching,
      submitButtonText,
      submitButtonClassName } = this.props

    const props = {
      fields,
      messages,
      events : {
        onSubmit          : this.handleFormSubmit,
        onValueChange     : this.handleChange,
        validateInput     : this.validateInput,
        handleKeyPress    : this.handleKeyPress,
        makeFieldTestable : this.makeFieldTestable
      },
      submitButtonData : {
        error,
        isFetching,
        submitButtonText,
        submitButtonClassName,
        onSubmit : this.handleFormSubmit
      }
    }

    return (
      <Form
        {...props}
      />
    )
  }

  /**
  * React Lifecycle Method: Executes right before the
  * component mounts
  */
  componentWillMount() {
  }
}
