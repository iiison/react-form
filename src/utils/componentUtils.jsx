import React from 'react'

export function drawElements ({ shouldUseDefaultClasses, classes, content, defaultClasses }) {
  return content
    ? <div className={`col-12 ${classes} ${shouldUseDefaultClasses && defaultClasses}`}>{content}</div>
    : ''
}

