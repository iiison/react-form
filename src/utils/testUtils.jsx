import React, { Component } from 'react'

export function wrapWithContext(context, contextTypes, Elements) {
  class WrapWithContext extends Component {
    getChildContext = () => context

    render() {
      return (
        <div>
          {this.props.children}
        </div>
      ) 
    }

    getChildContext() {
      return context
    }

    static childContextTypes = contextTypes
  }

  return WrapWithContext
}

