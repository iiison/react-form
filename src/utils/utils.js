/**
 * Debounce input function
 *
 * @param   {Function} func function to be debounced
 * @param   {Number}   time time to wait
 *
 * @returns {Function}      debounced function
 */
export function debounce(func, time) {
  let timeout

  return function(...args) {
    const ref = this

    clearTimeout(timeout)
    timeout = setTimeout(() => func.apply(ref, args), time)
  }
}

