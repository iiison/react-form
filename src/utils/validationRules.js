const validations = {
  email() {
    return {
      rule    : /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i,
      formatter (fieldName) {
        const message = '%c is not valid'

        return message.replace('%c', fieldName)
      }
    }
  },
  required() {
    return {
      rule    : /\S/,
      formatter (fieldName) {
        const message = '%c is required'

        return message.replace('%c', fieldName)
      }
    }
  }
}

export default validations
