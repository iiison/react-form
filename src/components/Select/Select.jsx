import React, { Component } from 'react'
import PropTypes from 'prop-types' // eslint-disable-line import/no-extraneous-dependencies
import { drawElements } from '$UTILS/componentUtils'

function getField({ props, context }) {
  const { id } = props
  const { formData : { fields } } = context

  return fields[id]
}

function getOptionValue(options, filterPropName, value) {
  return options.filter((option) => {
    return option[filterPropName] === value
  })[0]
}

/* eslint-disable jsx-a11y/click-events-have-key-events */
export default class Select extends Component {
  drawSelectedValue = () => {
    const field = getField({
      props   : this.props,
      context : this.context
    })
    const { placeholder, options, value, classes : { displayValueClass } } = field
    const content = value ? getOptionValue(options, 'value', value).displayName : placeholder

    return <div className={`col-12 field-value ${displayValueClass} select-value`}>{content}</div>
  }

  drawOptions = ({ optionClass }) => {
    // const { id } = this.props
    // const { fields } = this.context
    const field = getField({
      props   : this.props,
      context : this.context
    })
    const { options, value, placeholder, classes } = field
    const defaultOption = (
      <div
        key='default'
        role='button'
        onClick={() => { this.updateSelect() }}
        className={`${optionClass} option default`}
      >
        {placeholder}
      </div>
    )

    const renderedOptions = options.map((option) => {
      const { id, displayName, isDisabled = false, value : optionValue } = option

      return (
        <div
          key={id}
          role='button'
          onClick={() => { this.updateSelect(option) }}
          className={`${optionClass} option ${value === optionValue && classes.selectedOptionClass} ${isDisabled ? 'disabled-option' : ''}`}
        >
          {displayName}
        </div>
      )
    })

    return [defaultOption, ...renderedOptions]
  }

  updateSelect({ value } = { value : '' }) {
    const field = getField({
      props   : this.props,
      context : this.context
    })
    const { id, events } = field
    const {
      setFieldValue,
      validateForm,
      formData = {}
    } = this.context

    setFieldValue({
      field,
      event : {},
      value,
    }, () => { validateForm(id) })

    if (events && events.onChange && !formData.errors[id].length) {
      events.onChange({
        formData,
        setFieldValue
      })
    }
  }

  render() {
    const {
      formData = {}
    } = this.context
    const {
      fields,
      defaultClasses,
      errors : allErrors
    } = formData
    const { id } = this.props

    if ( fields && fields[id] ) {
      const field = fields[id]

      const {
        label,
        classes,
        shouldUseDefaultClasses
      } = field
      const {
        contClass,
        labelClass,
        errorClass,
        optionsContClass,
        optionClass
      } = classes
      const {
        contClass  : defaultContClass,
        errorClass : defaultErrorClass,
        labelClass : defaultLabelClass
      } = defaultClasses
      const errors = allErrors[id]

      return (
        <div className={`select-box ${contClass} ${shouldUseDefaultClasses && defaultContClass} grid col-12 input-cont`}>
          {
            drawElements({
              shouldUseDefaultClasses,
              classes        : `${labelClass} label`,
              content        : label,
              defaultClasses : defaultLabelClass
            })
          }
          <div className={`select`}>
            {this.drawSelectedValue()}
            <div className={`grid options ${optionsContClass}`}>
              {this.drawOptions({ optionClass })}
            </div>
          </div>
          {
            drawElements({
              shouldUseDefaultClasses,
              classes        : `${errorClass} error`,
              content        : errors,
              defaultClasses : defaultErrorClass
            })
          }
        </div>
      )
    }

    return (<div />)
  }

  componentDidMount() {
    this.context.addField(this.props)
  }

  static defaultProps = {
    value                   : '',
    label                   : '',
    events                  : {},
    validate                : '',
    placeholder             : 'Select',
    shouldValidateField     : true,
    shouldUseDefaultClasses : true,
    classes                 : {
      contClass           : '',
      labelClass          : '',
      fieldClass          : '',
      errorClass          : '',
      optionClass         : '',
      selectedOptionClass : ''
    }
  }

  static propTypes = {
    id                      : PropTypes.string.isRequired,
    value                   : PropTypes.string,
    label                   : PropTypes.string,
    validate                : PropTypes.string,
    shouldValidateField     : PropTypes.bool,
    shouldUseDefaultClasses : PropTypes.bool,
    placeholder             : PropTypes.string,
    displayName             : PropTypes.string.isRequired,
    options                 : PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
    classes                 : PropTypes.shape({
      contClass           : PropTypes.string,
      labelClass          : PropTypes.string,
      fieldClass          : PropTypes.string,
      errorClass          : PropTypes.string,
      optionClass         : PropTypes.string,
      selectedOptionClass : PropTypes.string
    }),
    events : PropTypes.shape({
      onChange : PropTypes.func
    })
  }

  static contextTypes = {
    addField      : PropTypes.func.isRequired,
    setFieldValue : PropTypes.func.isRequired,
    validateForm  : PropTypes.func.isRequired,
    formData      : PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
  }
}
/* eslint-enable */

