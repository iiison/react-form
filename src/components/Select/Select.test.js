import React               from 'react'
import renderer            from 'react-test-renderer'
import PropTypes           from 'prop-types'
import { mount }           from 'enzyme'

import Select               from './Select'
import FormContainer       from '../../FormContainer'
import { wrapWithContext } from '../../utils/testUtils'

/* eslint-disable react/jsx-filename-extension */
const defaultContext = {
  addField      : jest.fn(),
  setFieldValue : jest.fn(),
  validateForm  : jest.fn(),
  formData      : {
    defaultClasses : {
      contClass  : 'default-contClass',
      fieldClass : 'default-fieldClass',
      errorClass : 'default-errorClass',
      labelClass : 'default-labelClass'
    },
    fields : {
      selectBox : {
        value   : '',
        options : []
      }
    }
  }
}
const contextTypes = {
  addField      : PropTypes.func,
  setFieldValue : PropTypes.func,
  formData      : PropTypes.object,
  validateForm  : PropTypes.func,
}
const defaultArgs = {
  id                      : 'selectBox',
  value                   : '',
  options                 : [],
  validate                : 'required',
  placeholder             : 'Select Value',
  displayName             : 'Select Box',
  shouldUseDefaultClasses : true,
  classes                 : {
    contClass  : 'cont-class',
    fieldClass : 'field-class',
    errorClass : 'error-class',
    labelClass : 'label-class'
  },
}
const options = [
  {
    id          : 'apple',
    value       : 'applee',
    displayName : 'Fruit Apple'
  },
  {
    id          : 'orange',
    value       : 'orange',
    displayName : 'Fruit orange'
  },
  {
    id          : 'pineapple',
    value       : 'pineapple',
    displayName : 'Fruit Pineapple'
  }
]

describe('>>> Select Container -- Snapshot Test', () => {
  it('Matches the snapshot with default props', () => {
    const selectPops = {
      ...defaultArgs,
      options
    }
    const tree = renderer.create(
      <FormContainer>
        <Select {...selectPops} />
      </FormContainer>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Matches the snapshot with custom props', () => {
    const props = {
      ...defaultArgs,
      options,
      label               : 'Select Fruits',
      value               : 'pineapple',
      placeholder         : 'Select One...',
      displayName         : 'Fruits',
      shouldValidateField : true,
      classes             : {
        contClass           : 'select-cont',
        labelClass          : 'select-label',
        fieldClass          : 'select-box',
        errorClass          : 'select-error',
        optionClass         : 'select-option',
        selectedOptionClass : 'selected-value'
      }
    }

    const tree = renderer.create(
      <FormContainer>
        <Select {...props} />
      </FormContainer>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Matches the snapshot with error in select box', () => {
    const customContext = { ...defaultContext }
    const props = {
      ...defaultArgs,
      options,
      onFieldChange : jest.fn(),
      validate      : 'required'
    }

    customContext.formData.fields.selectBox = props
    customContext.formData.errors = {
      selectBox : 'some error'
    }

    const WrappedWithContextComponent = wrapWithContext(customContext, contextTypes)
    const tree = renderer.create(
      <WrappedWithContextComponent>
        <Select {...props} />
      </WrappedWithContextComponent>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
// *************************************************************

// Test DOM Behaviour
describe('>>> Select Box Container -- Shallow Rendering', () => {
  let mountedScreen
  const selectScreen = (customProps = {}) => {
    const initialState = {
      id    : 'selectBox',
      value : '',
      options
    }

    if (!mountedScreen) {
      const updatedState = {
        ...initialState,
        ...customProps
      }

      mountedScreen = mount(<FormContainer><Select {...updatedState} /></FormContainer>)
    }

    return mountedScreen
  }

  beforeEach(() => {
    mountedScreen = undefined
  })

  it('Triggers the change function when select value is changed', () => {
    const props = {
      events : {
        onChange : jest.fn()
      }
    }

    const tree = selectScreen(props)
    const optionsInTree = tree.find({role : 'button'})
    const optionOrange = optionsInTree.at(2)

    optionOrange.simulate('click')

    const expectedValue = 'Fruit orange'
    const selectedValue = tree
      .find('.field-value')
      .at(0)
      .text()

    expect(selectedValue).toEqual(expectedValue)
    expect(props.events.onChange).toHaveBeenCalled()
  })
})
/* eslint-enable*/

