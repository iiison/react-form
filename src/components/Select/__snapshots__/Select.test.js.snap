// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Select Container -- Snapshot Test Matches the snapshot with custom props 1`] = `
<form
  action=""
>
  <div
    className="select-box select-cont  grid col-12 input-cont"
  >
    <div
      className="col-12 select-label label "
    >
      Select Fruits
    </div>
    <div
      className="select"
    >
      <div
        className="col-12 field-value undefined select-value"
      >
        Fruit Pineapple
      </div>
      <div
        className="grid options undefined"
      >
        <div
          className="select-option option default"
          onClick={[Function]}
          role="button"
        >
          Select One...
        </div>
        <div
          className="select-option option false "
          onClick={[Function]}
          role="button"
        >
          Fruit Apple
        </div>
        <div
          className="select-option option false "
          onClick={[Function]}
          role="button"
        >
          Fruit orange
        </div>
        <div
          className="select-option option selected-value "
          onClick={[Function]}
          role="button"
        >
          Fruit Pineapple
        </div>
      </div>
    </div>
    
  </div>
</form>
`;

exports[`>>> Select Container -- Snapshot Test Matches the snapshot with default props 1`] = `
<form
  action=""
>
  <div
    className="select-box cont-class  grid col-12 input-cont"
  >
    
    <div
      className="select"
    >
      <div
        className="col-12 field-value undefined select-value"
      >
        Select Value
      </div>
      <div
        className="grid options undefined"
      >
        <div
          className="undefined option default"
          onClick={[Function]}
          role="button"
        >
          Select Value
        </div>
        <div
          className="undefined option false "
          onClick={[Function]}
          role="button"
        >
          Fruit Apple
        </div>
        <div
          className="undefined option false "
          onClick={[Function]}
          role="button"
        >
          Fruit orange
        </div>
        <div
          className="undefined option false "
          onClick={[Function]}
          role="button"
        >
          Fruit Pineapple
        </div>
      </div>
    </div>
    
  </div>
</form>
`;

exports[`>>> Select Container -- Snapshot Test Matches the snapshot with error in select box 1`] = `
<div>
  <div
    className="select-box cont-class default-contClass grid col-12 input-cont"
  >
    
    <div
      className="select"
    >
      <div
        className="col-12 field-value undefined select-value"
      >
        Select Value
      </div>
      <div
        className="grid options undefined"
      >
        <div
          className="undefined option default"
          onClick={[Function]}
          role="button"
        >
          Select Value
        </div>
        <div
          className="undefined option false "
          onClick={[Function]}
          role="button"
        >
          Fruit Apple
        </div>
        <div
          className="undefined option false "
          onClick={[Function]}
          role="button"
        >
          Fruit orange
        </div>
        <div
          className="undefined option false "
          onClick={[Function]}
          role="button"
        >
          Fruit Pineapple
        </div>
      </div>
    </div>
    <div
      className="col-12 error-class error default-errorClass"
    >
      some error
    </div>
  </div>
</div>
`;
